### PR TITLE
feat(query): Add optional column names to create/alter view statement

### LINKED ISSUE #6854: Feature: Make Databend's create view syntax closer to mysql
**Summary**
It can be parsed normally in mysql, but the following error is reported in databend
```
mysql> create view v1(a) AS select number from numbers(10);
ERROR 1105 (HY000): Code: 1005, displayText = sql parser error: need `AS` after VIEW NAME.
```

### PR DESCRIPTION
**Pull request by johnhaxx7**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add optional column names to create/alter view statement

CREATE VIEW [IF NOT EXISTS] [db.]view_name **[(\<column\>, ...)]** AS SELECT query
ALTER VIEW [db.]view_name **[(\<column\>, ...)]** AS SELECT query

so that column names can be renamed.

If column names is not empty, the subquery is rewrote to 
```
select * from ({subquery}) {viewname}( names ) 
``` 
When we select/describe from the view, it'll parse from the new subquery to ensure we get the right column names.

Thanks to @sundy-li for the idea!

Closes #6854

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_view_create.rs [hunk 1]
```diff
@@ -69,7 +70,28 @@ impl CreateViewInterpreter {
...
         let mut options = BTreeMap::new();
-        options.insert("query".to_string(), self.plan.subquery.clone());
+
+        let subquery = if self.plan.column_names.is_empty() {
+            self.plan.subquery.clone()
+        } else {
+            let mut planner = Planner::new(self.ctx.clone());
+            let (plan, _, _) = planner.plan_sql(&self.plan.subquery.clone()).await?;
+            if plan.schema().fields().len() != self.plan.column_names.len() {
+                return Err(ErrorCode::BadDataArrayLength(format!(
```
Comment (johnhaxx7) [inline src/query/service/src/interpreters/interpreter_view_create.rs:80]: Please advice if `BadDataArrayLength` is appropriate, thanks!

File: src/query/service/src/interpreters/interpreter_view_alter.rs [hunk 2]
```diff
@@ -86,7 +87,27 @@ impl AlterViewInterpreter {
...
+                    "column name length mismatch, expect {}, got {}",
+                    plan.schema().fields().len(),
+                    self.plan.column_names.len(),
+                )));
+            }
+            format!(
+                "select * from ({}) {}({})",
+                self.plan.subquery,
+                self.plan.viewname,
+                self.plan.column_names.join(",")
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_view_alter.rs:106]: `join(", ")`

File: src/query/service/src/interpreters/interpreter_view_create.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/service/src/interpreters/interpreter_view_create.rs:80]: It's ok

File: src/query/service/src/interpreters/interpreter_view_alter.rs (hunk 2 above)
Comment (johnhaxx7) [reply src/query/service/src/interpreters/interpreter_view_alter.rs]: Done!

**Review by sundy-li — APPROVED** [review]

**Comment by johnhaxx7** [comment]
Hi @BohuTANG , seems like I don't have access to merge, can you please help if possible? thanks!

**Pull request merged**